os/drivers/lcd/mipi_lcd.c

```diff
@@ -496,34 +505,60 @@ static int lcd_getpower(FAR struct lcd_dev_s *dev)
  *   Power off the LCD.
  * 	 Switch to command mode and display off, power off the LCD.
  *
+ * Returns:
+ * 	 OK if power on sequence succeeded,
+ * 	 ERROR if power on sequence failed.
+ *
  ****************************************************************************/
 
-static void lcd_power_off(FAR struct mipi_lcd_dev_s *priv)
+static int lcd_power_off(FAR struct mipi_lcd_dev_s *priv)
 {
+	DEBUGASSERT(priv);
+	if (priv->lcdonoff == LCD_OFF) {
+		lcddbg("ERROR: LCD already powered off\n");
+		return ERROR;
+	}
+
 	lcddbg("Switch to CMD mode\n");
 	priv->config->mipi_mode_switch(CMD_MODE);
-	priv->lcdonoff = LCD_OFF;
+
 	lcm_setting_table_t display_off_cmd = {0x28, 0, {0x00}};
-	send_cmd(priv, display_off_cmd);
-	/* The power off must operate only when LCD is ON */
+	if (send_cmd(priv, display_off_cmd) != OK) {
+		lcddbg("ERROR: LCD power off failed\n");
+		return ERROR;
```

Comment:
This will finally calls bsp function, and perhaps return value is one of value in errno.h
so you should return one of errno, not ERROR